### PR TITLE
Form woman coding teaser

### PR DESCRIPTION
**Pull request by M3kH**
Keep the love and the mutual respect here.
girl party

### CODE REVIEW

File: pages/apply/index.vue [hunk 1]
```diff
@@ -125,6 +126,37 @@ export default {
...
+                }
+            });
+            return true;
+        },
+        emptyRequiredField(e) {
+            e.parentNode.classList.add("active");
+            e.value = "";
+        },
+        successMSG() {
+            const { successMessage } = this.$refs;
```
[the remark this answers is not in the document]
Comment (gisaac85) [reply pages/apply/index.vue]: ok

File: pages/apply/index.vue [hunk 2]
```diff
@@ -125,6 +126,37 @@ export default {
...
+                this.$refs.phone,
+                this.$refs.education,
+                this.$refs.how_hear
+            ];
+            inputs.forEach(element => {
+                if (element.value === "" || element.value === null) {
+                    element.parentNode.classList.add("active");
+                    element.value = "Required field";
+                    return false;
+                } else {
```
[the remark this answers is not in the document]
Comment (gisaac85) [reply pages/apply/index.vue]: ok

File: pages/apply/index.vue [hunk 3]
```diff
@@ -125,6 +126,37 @@ export default {
...
+                this.$refs.city,
+                this.$refs.phone,
+                this.$refs.education,
+                this.$refs.how_hear
+            ];
+            inputs.forEach(element => {
+                if (element.value === "" || element.value === null) {
+                    element.parentNode.classList.add("active");
+                    element.value = "Required field";
+                    return false;
```
[the remark this answers is not in the document]
Comment (gisaac85) [reply pages/apply/index.vue]: yes

File: package.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
   },
   "scripts": {
     "postinstall": "cd api && npm i",
-    "dev": "export ENVIRONMENT=dev&& npm-run-all dev:config dev:start",
+    "dev": " npm-run-all dev:config dev:start",
```
[the remark this answers is not in the document]
Comment (Aya-Alabrash) [reply package.json]: No I try 😭

[the remark this answers is not in the document]
Comment (M3kH) [reply package.json]: Mhh.. we need to find a solution for this :-\ maybe using the `.env` file could work.

[the remark this answers is not in the document]
Comment (isaachinman) [reply package.json]: As suggested in the original PR, `better-run-npm` is good for these type of script-specific environments. We should use `.env` primarily for secrets.

File: pages/apply/index.vue [hunk 5]
```diff
@@ -110,13 +111,13 @@ export default {
...
-            let req = await axios.post(process.env.lambdaUrl + "apply", fields) // eslint-disable-line
-
-            // TODO: redirect to application success route
-            // TODO: try/catch for application POST failure
+            if (this.emptyInputs() === true) {
+                const fields = {};
+                Object.values(e.target.elements).forEach(
+                    input => (fields[input.name] = input.value)
+        ); // eslint-disable-line
+        let req = await axios.post(process.env.lambdaUrl + "apply", fields); // eslint-disable-line
```
Comment (isaachinman) [inline pages/apply/index.vue]: We need error handling and redirect to success route here, as indicated by my TODO comments which were deleted.

File: api/.env [hunk 6]
```diff
@@ -0,0 +1 @@
+MOLLIE_API_KEY="test_uRVRx9P8fjk9B5wJHuFFEyRuD2QaW3"
```
[the remark this answers is not in the document]
Comment (Aya-Alabrash) [reply api/.env]: done

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (gisaac85) [reply package.json]: so what is the last decision? 
write better-run-npm or .env ???

File: pages/apply/index.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (Aya-Alabrash) [reply pages/apply/index.vue]: it's solved now

File: pages/teaser/index.vue [hunk 7]
```diff
@@ -45,49 +97,116 @@ export default {
             teaser_about: teaser_about ? teaser_about : null
         };
     },
-    components: {}
+    mounted: function() {
+        this.$el.querySelectorAll(".input").forEach(i => {
+            if (i.value.length == 0) {
+                i.parentNode.classList.remove("active");
```
[the remark this answers is not in the document]
Comment (Aya-Alabrash) [reply pages/teaser/index.vue]: where is the repeated?

File: pages/teaser/index.vue [hunk 8]
```diff
@@ -45,49 +97,116 @@ export default {
...
+                this.$refs.education
+            ];
+            inputs.forEach(element => {
+                if (element.value === "" || element.value === null) {
+                    element.parentNode.classList.add("active");
+                    element.value = "Required field";
+                    return false;
+                } else {
+                    this.successMSG();
+                    return true;
```
[the remark this answers is not in the document]
Comment (Aya-Alabrash) [reply pages/teaser/index.vue]: okidokie

File: api/.env (hunk 6 above)
[the remark this answers is not in the document]
Comment (M3kH) [reply api/.env]: Lie :P is still here.

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (M3kH) [reply package.json]: My call goes to `.env`

[the remark this answers is not in the document]
Comment (isaachinman) [reply package.json]: In general, environment variables should only be in a `.env` file if they specifically need to be outside of version control. Every single developer who runs `yarn dev` will need `ENVIRONMENT=dev` and it is not sensitive data.

Imagine if you wanted to test the app in production, locally.

File: pages/apply/index.vue (hunk 5 above)
Comment (isaachinman) [reply pages/apply/index.vue]: @gisaac85 @Aya-Alabrash Either of you have a quick fix for this?

**Comment by mkruijt** [comment]
Hi @Aya-Alabrash and @gisaac85 I see some comments that you are fixing stuff, but I do not see any new commits?

**Pull request merged**